frontend/src/app/category-list/category-list.component.ts

```diff
@@ -0,0 +1,88 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { Category, CategoryBackendService } from "../services/category-backend.service";
+import { MatTable } from "@angular/material/table";
+import { KeycloakService } from "keycloak-angular";
+import { Router } from "@angular/router";
+import { MatDialog } from "@angular/material/dialog";
+
+import { MatSnackBar } from "@angular/material/snack-bar";
+import { DialogWindowComponent } from "../dialog-window/dialog-window.component";
+
+@Component({
+  selector: 'app-category-list',
+  templateUrl: './category-list.component.html',
+  styleUrls: ['./category-list.component.css']
+})
+export class CategoryListComponent implements OnInit {
+
+  categories: Category[] = [];
+
+  category: Category;
+
+  displayedColumns: string[] = ['id', 'name', 'description', 'actions']
+
+  public role: boolean;
+
+  @ViewChild(MatTable) table: any;
+
+  constructor(private categoryService: CategoryBackendService,
+              private keycloak: KeycloakService,
+              public readonly router: Router,
+              public dialog: MatDialog,
+              private snackBar: MatSnackBar) {
+  }
+
+  ngOnInit() {
+    this.getAllCategories();
+  }
+
+  getAllCategories() {
+    this.role = this.keycloak.isUserInRole('ROLE_USER');
+
+    this.categoryService.getAllCategories().subscribe((data) => {
+      this.categories = data;
+
+      this.table.renderRows();
+    })
+  }
+
+  getCategoryById(id: number) {
+    this.role = this.keycloak.isUserInRole('ROLE_ADMIN');
+
+    console.log(id);
+  }
+
+  createCategory() {
+    this.role = this.keycloak.isUserInRole('ROLE_ADMIN');
+  }
+
+  updateCategory(id: number) {
+    this.role = this.keycloak.isUserInRole('ROLE_ADMIN');
+
+    console.log(id);
+  }
+
+  deleteCategory(id: number) {
+    const dialogRef = this.dialog.open(DialogWindowComponent);
+
+    dialogRef.afterClosed().subscribe((res) => {
+      switch (res.event) {
+        case "confirm-option":
+          this.categoryService.deleteCategory(id).subscribe(() => {
+            },
+            (error) => {
+              this.snackBar.open(`${error.message}`, 'OK', {
+                duration: 3000
+              });
+            });
+
+          window.location.reload();
```

Comment:
never use `window.location.reload();`. this is not the way angular work. 

You must use next:

1. get id from `ActivatedRoute paramMap`.
2. Use `subscribe()` when want to call backend api. 
3. Unsubscribe for each `subscribe()` function using `.pipe(takeUntil(this.unsubscribe))`

```
    private unsubscribe: Subject<void> = new Subject();

    ngOnInit() {
        this.route.paramMap.pipe(
            switchMap((params: ParamMap) => of(Number(params.get('id')))),
            switchMap((id: number) => this.apiService.getSomethingById(id)),
            takeUntil(this.unsubscribe)
        ).subscribe(data => this.data = data);
    }

    ngOnDestroy() {
        this.unsubscribe.next();
        this.unsubscribe.complete();
    }
```
4. Use `this.router.navigate(['/admin/categories/${id}',]);` where `private router: Router`